lib/galaxy/tool_util/deps/containers.py

```diff
@@ -181,6 +194,19 @@ def __init__(self, app_info):
         self.enable_mulled_containers = app_info.enable_mulled_containers
         self.app_info = app_info
         self.container_resolvers = self.__build_container_resolvers(app_info)
+        self.mulled_resolution_cache = None
+        if app_info.mulled_resolution_cache_type:
+            # Do not want to make beaker a required dependency of container resolution, that doesn't make a lot
```

Comment:
Would you be opposed to making this a top-level import with try / except ImportError ?

Reply:
How about https://github.com/galaxyproject/galaxy/pull/8741/commits/b6b67f1f474e9d9cdeb07e06176338d493c86fb0 - cleaner imports, cleaner dependencies.

Reply:
😍